### ISSUE TITLE
Loading graphic

### ISSUE THREAD

**Issue by chrismytton** [everypolitician/membership_matrix#1]
With large countries it can take a long time to load the interface. Would be good to have some kind of loading graphic to show there's _something_ going on.

**Comment by tmtmtmtm** [comment]
is this purely a function of many memberships there are, or are there other factors in play? Would it help if it loaded the most recent term first, and then filled in others in the background, or anything like that?

**Comment by chrismytton** [comment]
Yep, I think it's because there are lots of memberships. However loading just the latest term first might be tricky (but certainly possible) as things currently work because we're using an html `<table>` so each _row_ is rendered together, one row for each area. So we'd have to do something clever with appending additional terms to existing rows for the correct areas for that to work.

**Comment by tmtmtmtm** [comment]
That doesn't seem like it should be _too_ tricky &lt;/handwave&gt; — and we'll probably need to work out how to do it anyway for countries where we have a lot of historic terms. I'd expect in those cases for only say 4 or 5 terms to be visible at once, and extra ones to get added on demand (either if you explicitly clicked something, or semi-magically as you scroll right or something). We certainly wouldn't want to have to wait for 114 columns to load once we have every US congress imported, for example…

**Comment by zarino** [comment]
Loading graphic is a really quick win here. (Just add a spinner gif as a background-image on the body, and make sure the table has a white background that covers it up once it's finished rendering).

Reducing loading time is, obviously, the long-term answer here. There might be ways we can optimise the JavaScript we've already written. Or, as @tmtmtmtm mentions, displaying fewer terms at a time might help. I've opened a new ticket for that one: https://github.com/everypolitician/membership_matrix/issues/2
